daemon/src/main/java/bisq/daemon/grpc/interceptor/GrpcCallRateMeter.java

```diff
@@ -1,95 +1,92 @@
-/*
- * This file is part of Bisq.
- *
- * Bisq is free software: you can redistribute it and/or modify it
- * under the terms of the GNU Affero General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or (at
- * your option) any later version.
- *
- * Bisq is distributed in the hope that it will be useful, but WITHOUT
- * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
- * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
- * License for more details.
- *
- * You should have received a copy of the GNU Affero General Public License
- * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
- */
-
 package bisq.daemon.grpc.interceptor;
 
-import bisq.common.Timer;
-import bisq.common.UserThread;
-
 import org.apache.commons.lang3.StringUtils;
 
+import java.util.ArrayDeque;
 import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
 
 import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
 
-import javax.annotation.Nullable;
-
 import static java.lang.String.format;
+import static java.lang.System.currentTimeMillis;
 
 @Slf4j
-public final class GrpcCallRateMeter {
+public class GrpcCallRateMeter {
 
     @Getter
     private final int allowedCallsPerTimeUnit;
     @Getter
     private final TimeUnit timeUnit;
     @Getter
-    private int callsCount = 0;
+    private final int numTimeUnits;
+
     @Getter
-    private transient boolean isRunning;
+    private transient final long timeUnitIntervalInMilliseconds;
 
-    @Nullable
-    private Timer timer;
+    private transient final ArrayDeque<Long> callTimestamps;
 
     public GrpcCallRateMeter(int allowedCallsPerTimeUnit, TimeUnit timeUnit) {
```

Comment:
`allowedCallsPerTimeUnit` is a bit misleading as it's really allowedCallsPerTimeWindow.

Reply:
Fixed in commit 10727fc0830330e004351cd0567b994173a5d5c5.